megamek/src/megamek/common/Terrain.java

```diff
@@ -370,6 +370,22 @@ public int movementCost(Entity e) {
             }
             return 0;
         case Terrains.RUBBLE:
+            if (e.getCrew().getOptions().booleanOption(OptionsConstants.PILOT_TM_MOUNTAINEER)) {
+                if (level == 6) {
+                    if (((e instanceof Mech) && ((Mech)e).isSuperHeavy())
+                        || (e.getCrew().getOptions().booleanOption(OptionsConstants.INFANTRY_FOOT_CAV)
+                                && (moveMode == EntityMovementMode.INF_LEG))) {
+                        return 0;
+                    }
+                    return 1;
+                }
+                if (((e instanceof Mech) && ((Mech)e).isSuperHeavy())
+                    || (e.getCrew().getOptions().booleanOption(OptionsConstants.INFANTRY_FOOT_CAV)
+                            && (moveMode == EntityMovementMode.INF_LEG))) {
+                    return 0;
+                }
+                return 0;
+            }
```

Comment:
I think it would be easier to write something like
```
int mountaineerModifier = 0;
if (mountaineer) {
    mountaineerModifier = 1;
}
```
And then always subtract the mountaineer Modifier.  This requires 4 new lines, and then modifier old lines, instead of adding a big block of mostly duplicated code.

Reply:
One of the reasons I asked if movecost should get a rewrite. It would be easier to make all of rely on a modifier. At the top of the switch add say int mp = 0; then in each case add the TO per hex cost. so for case WOODS use mp += level then subtract each possible modifier from it and return the final value.

Reply:
I don't generally get the full context looking at diffs.  Yea, I can expand them, but you have to expand a lot.  I think, overall, the structure is probably okay.  It's not really dense code, but it just has a lot of white space.  In this particular case, you duplicated a big block of code, and I think it could be written without any real duplication.